core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/chunkedencoding/ChunkEncodedInputStreamTest.java

```diff
@@ -0,0 +1,225 @@
+package software.amazon.awssdk.http.auth.aws.chunkedencoding;
+
+import static java.util.Arrays.copyOf;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.deriveSigningKey;
+import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.hash;
+import static software.amazon.awssdk.utils.BinaryUtils.toHex;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.http.auth.aws.internal.signer.RollingSigner;
+import software.amazon.awssdk.http.auth.aws.signer.CredentialScope;
+import software.amazon.awssdk.http.auth.aws.util.SignerConstant;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.utils.Pair;
+
+public class ChunkEncodedInputStreamTest {
```

Comment:
Can we add tests to cover the cases where multiple extensions, trailers are provided?